src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs

```diff
@@ -537,6 +554,403 @@ public CorElementType PeekElementType()
+            if ((methodFlags & (uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_MemberRefToken) != 0)
+            {
+                methodFlags &= ~(uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_MemberRefToken;
+                result = ParseMethodRefToken(owningTypeOverride: owningTypeOverride);
+            }
+            else
+            {
+                result = ParseMethodDefToken(owningTypeOverride: owningTypeOverride);
+            }
+
+            if ((methodFlags & (uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_MethodInstantiation) != 0)
+            {
+                methodFlags &= ~(uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_MethodInstantiation;
+                uint typeArgCount = ReadUInt();
+                TType[] instantiationArgs = new TType[typeArgCount];
+                for (int typeArgIndex = 0; typeArgIndex < typeArgCount; typeArgIndex++)
+                {
+                    instantiationArgs[typeArgIndex] = ParseType();
+                }
+                result = _provider.GetInstantiatedMethod(result, instantiationArgs.ToImmutableArray());
+            }
+
+            if ((methodFlags & (uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_Constrained) != 0)
+            {
+                methodFlags &= ~(uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_Constrained;
+                result = _provider.GetConstrainedMethod(result, ParseType());
+            }
+
+            // Any other flags should just be directly recorded
+            if (methodFlags != 0)
+                result = _provider.GetMethodWithFlags((ReadyToRunMethodSigFlags)methodFlags, result);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Read a methodDef token from the signature and output the corresponding object to the builder.
+        /// </summary>
+        /// <param name="builder">Output string builder</param>
+        private TMethod ParseMethodDefToken(TType owningTypeOverride)
+        {
+            uint rid = ReadUInt();
+            return _provider.GetMethodFromMethodDef(_metadataReader, MetadataTokens.MethodDefinitionHandle((int)rid), owningTypeOverride);
+        }
+
+
+        /// <summary>
+        /// Read a memberRef token from the signature and output the corresponding object to the builder.
+        /// </summary>
+        /// <param name="builder">Output string builder</param>
+        /// <param name="owningTypeOverride">Explicit owning type override</param>
+        private TMethod ParseMethodRefToken(TType owningTypeOverride)
+        {
+            uint rid = ReadUInt();
+            return _provider.GetMethodFromMemberRef(_metadataReader, MetadataTokens.MemberReferenceHandle((int)rid), owningTypeOverride);
+        }
+
+    }
+    public class TextSignatureDecoderContext
+    {
+        public TextSignatureDecoderContext(IAssemblyResolver options)
+        {
+            Options = options;
+        }
+
+        public StringBuilder CurrentBuilder;
```

Comment:
Should we make this field public?

Reply:
We don't actually need this field. I'll delete it.